src/store/getters/state-getters.ts

```diff
@@ -1029,3 +1060,13 @@ export const getSpecialResolutionConfirmValid = (state: StateIF): boolean => {
 export const getRestoration = (state: StateIF): RestorationStateIF => {
   return state.stateModel.restoration
 }
+/** Returns true when users can change the sole proprietor (SP).
+ * Restricts the ability of non-staff users from changing the sole
+ * proprietor when the SP is a non-individual.  This restriction has been
+ * added to ensure that changes made in business-edit-ui are also updated by
+ * staff in COLIN */
+export const showSoleProprietorChangeButton = (state: StateIF, getters): boolean => {
```

Comment:
Since this getter is only used in 1 place, can it moved to that component (or its mixin)?

Reply:
My recommendation would be that all business logic belongs in the store.  Placing the logic in a component -- even if it's only used once -- means that we're more likely to create a duplicate when we need identical logic for another component.   I'm not crazy about the idea of extending mixins after you said that they weren't supported in Vue3.

Reply:
I might be wrong on the Vue3 mixin thing. Maybe they just need to be handled differently.

I generally agree with your point, but the "should I show this edit button" logic is very specific to the ListPeopleAndRoles component, so I still think it should be moved there instead of in the store. Not a deal breaker though.

Reply:
Let's discuss the business logic issue further.  It would be great to have a clear vision of the end goal.

Reply:
I'd like to see a code architecture where it's easy to create / drop in a new component without having to know much about the architecture, and it's easy to create a new thing (eg, filing) by simply putting existing components on a new page (view).

I'd like to see a relatively "dumb" UI, with as little business logic as possible.

And I like the idea of "independent" components... Maybe call them "smart" components (vs dumb), where most of what they do is localized and their interface to the rest of the system is clear / well defined. (I know this seems to go against the "store does everything" model. But maybe there's nice middle ground.)

I like the "tier" idea and separation of concerns, which to me keeps things independent and well defined. (It also makes unit testing easier.)

What do you think?

Reply:
I think the architecture that I wrote for Roadsafety is what you're describing.  Form components (made up of one or more questions) could dropped into any new form and the only wiring required was the store location and the display options.  They had local smarts.  For example, they could display a spinner when busy but the component logic was encapsulated. The parent didn't need to know how the inner workings behaved and new fields could be added the shared component without changing the parent. I'll tee-up a meeting for us to discuss the implementation details.

Reply:
Jonathan and I discussed this larger architectural question in various 1:1 meetings (esp in the context of the new Consent for Continuation Out filing, and re: app startup in App.vue, and mixins vs store modules).

Anyway, I'm OK with the implementation above since it's a temporary change, and long-term we'll figure out our ideal model so let's not get side tracked by that right now.

Resolving this conversation.